diff --git a/Makefile b/Makefile
index be56586..6fe6709 100644
--- a/Makefile
+++ b/Makefile
@@ -322,13 +322,24 @@ $(dpi-library)/ariane_dpi.so: $(dpi)
 # if you want to run in batch mode, use make <testname> batch-mode=1
 # alternatively you can call make sim elf-bin=<path/to/elf-bin> in order to load an arbitrary binary
 generate-trace-vsim:
-       make sim preload=$(preload) elf-bin= batch-mode=1
+       make sim elf-bin=$(elf-bin) batch-mode=1
        make generate-trace
 
+#sim: build
+#      vsim${questa_version} +permissive $(questa-flags) -lib $(library) +MAX_CYCLES=$(max_cycles) +UVM_TESTNAME=$(test_case) \
+#      -wlf $(preload).wlf -do "coverage save -onexit $(preload).ucdb; run -a; quit -code [coverage attribute -name TESTSTATUS -concise -value $(preload)]" $(questa-cmd) \
+#      +BASEDIR=$(riscv-test-dir) $(uvm-flags) $(QUESTASIM_FLAGS) -gblso $(SPIKE_ROOT)/lib/libfesvr.so -sv_lib $(dpi-library)/ariane_dpi  \
+#      ${top_level}_optimized +permissive-off ++$(elf-bin) ++$(target-options) | tee sim.log
+
+#sim: build
+#      vsim${questa_version} +permissive $(questa-flags) $(questa-cmd) -lib $(library) +MAX_CYCLES=$(max_cycles) +UVM_TESTNAME=$(test_case) \
+#      +BASEDIR=$(riscv-test-dir) $(uvm-flags) $(QUESTASIM_FLAGS) -Lf $(RISCV)/lib/libfesvr.a -sv_lib $(dpi-library)/ariane_dpi  \
+#      ${top_level}_optimized +permissive-off ++$(elf-bin) ++$(target-options) | tee sim.log
+
 sim: build
        vsim${questa_version} +permissive $(questa-flags) -lib $(library) +MAX_CYCLES=$(max_cycles) +UVM_TESTNAME=$(test_case) \
-       -do "coverage save -onexit $(preload).ucdb; run -a; quit -code [coverage attribute -name TESTSTATUS -concise -value $(preload)]" $(questa-cmd) \
-       +BASEDIR=$(riscv-test-dir) $(uvm-flags) $(QUESTASIM_FLAGS) -gblso $(SPIKE_ROOT)/lib/libfesvr.so -sv_lib $(dpi-library)/ariane_dpi  \
+       -wlf $(elf-bin).wlf -do " log -r /*; add wave -r /*;run -a;quit" \
+       +BASEDIR=$(riscv-test-dir) $(uvm-flags) $(QUESTASIM_FLAGS)  -Lf $(SPIKE_ROOT)/lib/libfesvr.a -sv_lib $(dpi-library)/ariane_dpi  \
        ${top_level}_optimized +permissive-off ++$(elf-bin) ++$(target-options) | tee sim.log
 
 $(riscv-asm-tests): build
